File: sdks/java-v2/src/test/java/com/dropbox/sign/api/FaxApiTest.java
package com.dropbox.sign.api;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.mockito.ArgumentMatchers.any;
import static org.mockito.ArgumentMatchers.eq;

import com.dropbox.sign.ApiClient;
import com.dropbox.sign.TestHelper;
import com.dropbox.sign.model.*;
import com.fasterxml.jackson.databind.JsonNode;
import java.io.File;
import java.util.Arrays;
import java.util.HashMap;
import org.junit.jupiter.api.Test;
import org.mockito.Mockito;

public class FaxApiTest {
    @Test
    public void faxSendTest() throws Exception {
        JsonNode expectedResponseData =
                TestHelper.getJsonContents(FaxGetResponse.class.getSimpleName());
        FaxGetResponse expectedResponse =
                FaxGetResponse.init(expectedResponseData.toString());
        ApiClient apiClient = TestHelper.setUpMock(200, expectedResponse);

        JsonNode requestData =
                TestHelper.getJsonContents(
                        FaxSendRequest.class.getSimpleName());

        FaxSendRequest request =
                FaxSendRequest.init(requestData.toString());
        request.addFilesItem(new File("test_fixtures/pdf-sample.pdf"));

        FaxApi api = new FaxApi(apiClient);
        FaxGetResponse response = api.faxSend(request);

        assertEquals(expectedResponse, response);
    }

    @Test
    public void faxGetTest() throws Exception {
        String faxId = "c2e9691c85d9d6fa6ae773842e3680b2b8650f1d";

        JsonNode expectedResponseData =
                TestHelper.getJsonContents(FaxGetResponse.class.getSimpleName());
        FaxGetResponse expectedResponse =
                FaxGetResponse.init(expectedResponseData.toString());
        ApiClient apiClient = TestHelper.setUpMock(200, expectedResponse);

        FaxApi api = new FaxApi(apiClient);
        FaxGetResponse response = api.faxGet(faxId);

        assertEquals(expectedResponse, response);
    }

    @Test
    public void faxListTest() throws Exception {
        JsonNode expectedResponseData =
                TestHelper.getJsonContents(FaxListResponse.class.getSimpleName());
        FaxListResponse expectedResponse =
                FaxListResponse.init(expectedResponseData.toString());
        ApiClient apiClient = TestHelper.setUpMock(200, expectedResponse);

        FaxApi api = new FaxApi(apiClient);
        FaxListResponse response = api.faxList();

        assertEquals(expectedResponse, response);
    }
}

File: sdks/php/test/Api/FaxApiTest.php
<?php

declare(strict_types=1);

namespace Dropbox\Sign\Test\Api;

use Dropbox\Sign\Api;
use Dropbox\Sign\Configuration;
use Dropbox\Sign\Model;
use Dropbox\Sign\Test\SignTestCase;
use Dropbox\Sign\Test\TestUtils;
use GuzzleHttp;
use SplFileObject;

class FaxApiTest extends SignTestCase
{
    /** @var Api\FaxApi */
    protected $api;

    protected function setUp(): void
    {
        parent::setUp();

        $this->client = new GuzzleHttp\Client([
            'handler' => GuzzleHttp\HandlerStack::create($this->handler),
        ]);

        $this->api = new Api\FaxApi(
            Configuration::getDefaultConfiguration(),
            $this->client,
        );
    }

    public function testFaxSend()
    {
        $requestClass = Model\FaxSendRequest::class;
        $requestData = TestUtils::getFixtureData($requestClass)['default'];

        $responseClass = Model\FaxGetResponse::class;
        $responseData = TestUtils::getFixtureData($responseClass)['default'];

        $this->setExpectedResponse($responseData);

        $obj = Model\FaxSendRequest::init(
            $requestData
        );
        $obj->setFiles([
            new SplFileObject(self::ROOT_FILE_PATH . '/pdf-sample.pdf'),
        ]);

        $response = $this->api->faxSend($obj);
        $serialized = TestUtils::toArray($response);

        $this->assertInstanceOf($responseClass, $response);
        $this->assertEquals($responseData, $serialized);
        $this->assertEquals($responseData, TestUtils::toArray($response));
    }

    public function testFaxList()
    {
        $responseClass = Model\FaxListResponse::class;
        $responseData = TestUtils::getFixtureData($responseClass)['default'];

        $this->setExpectedResponse($responseData);

        $response = $this->api->faxList();
        $serialized = TestUtils::toArray($response);

        $this->assertInstanceOf($responseClass, $response);
        $this->assertEquals($responseData, $serialized);
        $this->assertEquals($responseData, TestUtils::toArray($response));
    }

    public function testFaxGet()
    {
        $faxId = 'c2e9691c85d9d6fa6ae773842e3680b2b8650f1d';

        $responseClass = Model\FaxGetResponse::class;
        $responseData = TestUtils::getFixtureData($responseClass)['default'];

        $this->setExpectedResponse($responseData);

        $response = $this->api->faxGet($faxId);
        $serialized = TestUtils::toArray($response);

        $this->assertInstanceOf($responseClass, $response);
        $this->assertEquals($responseData, $serialized);
        $this->assertEquals($responseData, TestUtils::toArray($response));
    }
}
